docs/DeveloperGuide.md

```diff
@@ -1,3 +1,1140 @@
+information, optimised for use through the command line interface. This
+app is for the head nurse of a hospital, if the user can type fast, it
+is better than a traditional GUI app.
+
+### **Purpose and scope**
+
+The purpose of this developer guide is to describe the architecture and
+software design decisions for our application. This guide will cover our
+program architecture, the logical view of major components and how our
+functions work.
+
+**Setting Up and Getting Started**
+----------------------------------
+
+### **Setting up**
+
+There are 2 prerequisites for this project
+
+1.  Java 11
+
+-   Launch your terminal and type "java -version" to ensure the correct
+    > version
+
+2.  Intellij IDEA
+
+### **Getting Started**
+
+1.  Fork this repo, and clone the fork into your computer.
+
+2.  Import the project as a gradle project
+
+3.  Run Duke.main and try a few commands
+
+4.  Run the tests to ensure they all pass.
+
+**Design**
+----------
+
+**Architecture**
+----------------
+
+![](media/image2.png){width="6.267716535433071in"
+height="5.638888888888889in"}
+
+Our application utilises many layers of abstraction which allows each
+individual component to be self contained yet able to work with other
+components. The Main Menu component allows direct access to other
+components as shown in the diagram.
+
+Shared functionalities are placed within the Common Classes component
+which will be elaborated on in the section [[Common
+Classes]{.ul}](#common-classes)
+
+### 
+
+### **UI component**
+
+1.  Each class has a specified UI class that serves the specific need
+    > for the individual class
+
+2.  There is a Main UI Class that shares all common UI functionalities
+    > amongst the classes. This will be elaborated in the Common Classes
+    > component
+
+> (Insert UI UML diagrams)
+
+### **Logic component**
+
+1.  This component would be the backbone of our program, where the
+    > functions of each individual class are crafted and contribute to
+    > the program.
+
+2.  Command is passed through the Parser.java class which determines
+    > what actions to perform. The actions to be performed are methods
+    > called in the classes stated below.
+
+[Actions:]{.ul}
+
+1.  In the menu of Doctor's Appointment, command of "add D12345 Alex M
+    > 21012021".
+
+2.  Input is passed along to the Parser.java class where the keyword
+    > "add" is detected.
+
+3.  addAppointment() method in the AppointmentActions.java class is
+    > called and adds the details into the list
+
+[Classes:]{.ul}
+
+1.  AppointmentActions
+
+2.  DrugActions
+
+3.  NurseScheduleActions
+
+4.  PatientList
+
+5.  StaffList
+
+### **Object Class component**
+
+This component is to distinguish the separate object that has been
+instantiated to hold the details of our target users. An object class
+allows for a structure that helps store details from user input in the
+format that we want to be saved.
+
+[Objects:]{.ul}
```

Comment:
Not sure what this is for - is it to control the css formatting? It seems to be rendering on the page directly, though
![image](https://user-images.githubusercontent.com/57757704/113082005-4df38580-920c-11eb-9494-52eb4740ab2c.png)